frontend/src/core/term/components/Term.js

```diff
@@ -0,0 +1,68 @@
+/* @flow */
+
+import * as React from 'react';
+
+import './Term.css';
+
+import type { TermType } from 'core/api';
+
+type Props = {|
+    term: TermType,
+    isReadOnlyEditor: boolean,
+    addTextToEditorTranslation: (string) => void,
+|};
+
+
+/**
+ * Show term entry with its metadata.
+ */
+export default class Term extends React.Component<Props> {
```

Comment:
Not really. I thought I'd use a Class Component since the component doesn't only render, but also handle events (and thus require additional methods). I see now that we have the exact same situation in the `AddComment.js` component.

Should the rule be to always use a functional component, unless we need State or Lifecycle? Although even those no longer seem to require Class Components. Should we then always use functional components?

Reply:
You'll see with my Editor refactor that the rule should be: use functional components. And that's it. :smile: With hooks it's possible to do everything that we could do with class components, so really there's no reason to use those anymore.